### PR TITLE
Split out FormattedNumberInput from Form.Item in SpecificLimitModal

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

This splits out the form into the desired format, ready to be used in code changes.

This also fixes a bug where you can submit and break the form with non numeric characters (doesn't update fast enough to catch in validation)

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1850** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1850" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1850?utm_source=stack-comment).

File: src/components/v2/shared/DistributionSplitsSection/SpecificLimitModal.tsx
Comment (wraeth-eth) [inline src/components/v2/shared/DistributionSplitsSection/SpecificLimitModal.tsx:54]: This is actually the original fix and reasoning that started me going down this rabbit hole. As I didn't want to keep the duplicate code (FormattedNumberInputNew), the rest of the stack unfolded

File: src/components/inputs/FormattedNumberInputNew.tsx
Comment (wraeth-eth) [inline src/components/inputs/FormattedNumberInputNew.tsx:7]: This is a transitory function, just used to make bridging the changes (without having broken code)

Comment (wraeth-eth) [reply src/components/inputs/FormattedNumberInputNew.tsx:7]: This is eventually renamed to `FormattedNumberInput` once the old one is finally discarded in the final PR

**Review by tomquirk — APPROVED** [review]

**Comment by wraeth-eth** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1850?utm_source=stack-comment).


Job Reference: Bz06LeDtEbubKjTKJ7Uz

**Pull request merged**
